pkg/service/requestroutersvc/service.go

```diff
@@ -29,6 +30,15 @@ func (s *Service) Route(ctx context.Context, tg target.Group) (*RouteResult, err
 	if err != nil {
 		return nil, err
 	}
+	//check to see if target group has a routed handler
+	//fetching resources will fail unless the tg is linked with a handler
+
+	log := logger.Get(ctx)
+	if err == ddb.ErrNoItems || len(routes.Result) < 1 {
+		//dont attempt refresh with no routes
+		log.Infow("no valid routes found when attempting to refetch target group resources", "target group:", tg.ID)
```

Comment:
This error message should go in the cache sync code not in the request router.

In the cache sync, you can check if the error is ErrNoValidRoute and then log any debug message that you need